app/src/main/java/com/example/bar/foo/myapplication/MainActivity.java

```diff
@@ -17,22 +21,43 @@
         System.loadLibrary("native-lib");
     }
 
+
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
         Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
         setSupportActionBar(toolbar);
 
-        FloatingActionButton fab = (FloatingActionButton) findViewById(R.id.fab);
-        fab.setOnClickListener(new View.OnClickListener() {
+        final Timer timer = new Timer();
+        final TimerTask timerTask = new TimerTask() {
+            @Override
+            public void run() {
+                farmPantsu();
+                //updateStatus(); how the heck do threads work again
```

Comment:
Was it getting upset about the UI being updated on a background thread?  Try `runOnUiThread`.